src/machine/usb_nxpmk66f18.go

```diff
@@ -0,0 +1,626 @@
+// +build nxp,mk66f18
+
+package machine
+
+import (
+	"device/nxp"
+	"fmt"
+	"reflect"
+	"runtime/interrupt"
+	"runtime/volatile"
+	"unsafe"
+)
+
+const debugUSB = true
+
+var USB0 = USB{USB0_Type: nxp.USB0, SCGC: &nxp.SIM.SCGC4, SCGCMask: nxp.SIM_SCGC4_USBOTG, ENDPT: (*[16]register8Aligned32)(unsafe.Pointer(&nxp.USB0.ENDPT0))}
+
+//go:align 512
+var usbBufferDescriptorRecords [usb_MaxBDTSize]usbBufferDescriptor
+var usbBufferDescriptorTable = usbBufferDescriptorTable_t{records: &usbBufferDescriptorRecords}
+
+func init() {
+	USB0.Interrupt = interrupt.New(nxp.IRQ_USB0, USB0.serviceInterrupt)
+}
+
+//go:linkname millisSinceBoot runtime.millisSinceBoot
+func millisSinceBoot() uint64
+
+type USB struct {
+	*nxp.USB0_Type
+	SCGC      *volatile.Register32
+	SCGCMask  uint32
+	Interrupt interrupt.Interrupt
+	ENDPT     *[16]register8Aligned32
+
+	CDCRXBuffer RingBuffer
+	Configured  bool
+
+	Log []string
+}
+
+func (u *USB) logf(format string, args ...interface{}) {
+	u.Log = append(u.Log, fmt.Sprintf(format, args...))
+}
+
+func (u *USB) debugf(format string, args ...interface{}) {
+	if debugUSB {
+		u.logf(format, args...)
+	}
+}
+
+func (u *USB) DumpLog() []string {
+	var log []string
+
+	state := interrupt.Disable()
+	log, u.Log = u.Log, nil
+	interrupt.Restore(state)
+
+	if log == nil {
+		return []string{}
+	}
+	return log
+}
```

Comment:
This looks like debug code. While functions that aren't called won't take up extra space (they get optimized away), the `Log` buffer cannot be optimized away. Maybe you can make it a separate global (separate from the `USB` type) if you want to keep it?